### DISCUSSION TITLE
Function "format" not found

### DISCUSSION THREAD

**Discussion by firecat888** [johnkerl/miller#941] [Q&A] [answered]
When I try to use function "format" in put verb,  it reported that "mlr: function name not found: format".
But the function "format" exists in the document [here](https://miller.readthedocs.io/en/latest/reference-dsl-builtin-functions/#format).
My command example:
`mlr put '$Subject=format("[{}]{}",${ID},$Subject)' example.csv  `

**Comment by johnkerl** [comment, marked as answer]
Hi @firecat888 ! The docs at [https://miller.readthedocs.io/en/latest/](https://miller.readthedocs.io/en/latest/) are for latest/HEAD while I imagine you're using the latest released version which is 6.0.0. And `format` was added after the 6.0.0 release.

This is confusing ... I put "Miller latest documentation" as the title at [https://miller.readthedocs.io/en/latest/](https://miller.readthedocs.io/en/latest/) but it's still a bit weird -- when you install Miller from a package manager like Brew/Conda/Chocolatey/etc you'll get the _release_ program version (like 6.0.0) by default and you can download/install the _latest_ program version if you choose to. Whereas you get the [_latest_ docs version](https://miller.readthedocs.io/en/latest) by default and you can click through to [_release_ doc versions](https://miller.readthedocs.io/en/latest/release-docs/index.html) if you choose to.

Anyway, for the present moment -- you can get a latest program version here: [https://miller.readthedocs.io/en/latest/installing-miller/](https://miller.readthedocs.io/en/latest/installing-miller/).

And I hope to release 6.1.0, with `format` and other nice new things, hopefully in a few weeks ...

**Comment by firecat888** [comment]
Hi @johnkerl  Thanks for you kindly response.  I got it.  By the way, Miller is a great tool and thanks for your great work.
